src/transforms/aws_cloudwatch_logs_subscription_parser.rs

```diff
@@ -0,0 +1,260 @@
+use super::Transform;
+use crate::{
+    config::{log_schema, DataType, TransformConfig, TransformContext, TransformDescription},
+    event::Event,
+    internal_events::{
+        AwsCloudwatchLogsSubscriptionParserEventProcessed,
+        AwsCloudwatchLogsSubscriptionParserFailedParse,
+    },
+};
+use chrono::{serde::ts_milliseconds, DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use std::iter;
+use string_cache::DefaultAtom as Atom;
+
+#[derive(Deserialize, Serialize, Debug, Clone, Derivative)]
+#[serde(deny_unknown_fields, default)]
+#[derivative(Default)]
+pub struct AwsCloudwatchLogsSubscriptionParserConfig {
+    pub field: Option<Atom>,
+}
+
+inventory::submit! {
+    TransformDescription::new::<AwsCloudwatchLogsSubscriptionParserConfig>("aws_cloudwatch_logs_subscription_parser")
+}
+
+#[typetag::serde(name = "aws_cloudwatch_logs_subscription_parser")]
+impl TransformConfig for AwsCloudwatchLogsSubscriptionParserConfig {
+    fn build(&self, _cx: TransformContext) -> crate::Result<Box<dyn Transform>> {
+        Ok(Box::new(AwsCloudwatchLogsSubscriptionParser::from(
+            self.clone(),
+        )))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "aws_cloudwatch_logs_subscription_parser"
+    }
+}
+
+#[derive(Debug)]
+pub struct AwsCloudwatchLogsSubscriptionParser {
+    field: Atom,
+}
+
+impl From<AwsCloudwatchLogsSubscriptionParserConfig> for AwsCloudwatchLogsSubscriptionParser {
+    fn from(
+        config: AwsCloudwatchLogsSubscriptionParserConfig,
+    ) -> AwsCloudwatchLogsSubscriptionParser {
+        let field = if let Some(field) = &config.field {
+            field
+        } else {
+            &log_schema().message_key()
+        };
+        AwsCloudwatchLogsSubscriptionParser {
+            field: field.clone(),
+        }
+    }
+}
+
+impl Transform for AwsCloudwatchLogsSubscriptionParser {
+    fn transform(&mut self, _event: Event) -> Option<Event> {
```

Comment:
This one is weird. This transform needs to create multiple events from a single input event, which only seems possible through `transform_into`, but I still have to implement `transform` even though it'll never be called.

cc/ @lukesteensen as this might be relevant to transforms part of the architecture revisit https://github.com/timberio/vector/blob/master/rfcs/2020-06-18-2625-architecture-revisit.md#transforms

Reply:
Yes, this is something I'd like to clean up by having more specific types of transforms. For now, this is the right workaround.